### DISCUSSION TITLE
send files after upload

### DISCUSSION THREAD

**Discussion by camelCaseNick** [cinnyapp/cinny#2817] [Feature Requests, Ideas]
### Describe the problem

Currently, there is a send button only visible after all uploads are done. If upload takes too long, you might see yourself doing other things, and forgetting to send.

### Describe the solution you'd like

I would like to have a way to mark the files for being sent after their upload is done, e.g. by providing the send button even while the upload is ongoing.

### Alternatives considered

Using an alarm clock app to set a 30 s timer?

**Comment by u9000-Nine** [comment]
So the current behavior is intended because it allows you to send a message and file at the same time. What are your thoughts on this?

**Reply by camelCaseNick** [reply]
Instead, a good UX would be to prompt users to write the message alongside the upload and only send it after the upload is done. There is no need to let a user wait for upload. There *is* a reason to let the send process wait.

Other clients (both matrix and different IM protocolls) let users select a file for upload, then let users write a message/caption after which a message bubble/entry appears with an upload indicator (e.g. loading icon or progress bar).

That is by far a better UX than the current one of Cinny which does not group the workflow around the user but around the protocol.
